Form/JQuery/Type/Select2Type.php

```diff
@@ -89,13 +90,16 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function getParent()
     {
-        return $this->widget;
+        if (class_exists('Symfony\Component\Form\Extension\Core\Type\\' . ucfirst($this->widget) . 'Type')) {
+            return 'Symfony\Component\Form\Extension\Core\Type\\' . ucfirst($this->widget) . 'Type';
+        }
+        return 'Symfony\Bridge\Doctrine\Form\Type\\' . ucfirst($this->widget) . 'Type';
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getName()
+    public function getBlockPrefix()
```

Comment:
I see, unfortunately this wouldn't work for Symfony 3.0, cf. the snipped below which is extracted from https://github.com/symfony/symfony/blob/master/UPGRADE-3.0.md :

=================
Passing type instances to Form::add(), FormBuilder::add() and the FormFactory::create*() methods is not supported anymore. Pass the fully-qualified class name of the type instead.

Before:

    $form = $this->createForm(new MyType());

After:

    $form = $this->createForm(MyType::class);
=================

So we can't use the current magic constructor to register different types from the same class anymore :/